### ISSUE TITLE
[Accessibility Audit] Search: Code Monitoring - Create/Edit

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#34195]
### Use cases to replicate

Taken from the [user journey list document](https://docs.google.com/document/d/1IA90OFew5MTcxFndKS2Xn4vb8sOt68sj537P7AGj_Tk/edit?usp=sharing). Use this for further context.

Please take these following use cases and replicate them when running your audit.

- On the create and edit code monitor page, I can navigate back to the code monitors landing page by using the code monitor icon
- On the create and edit code monitor page, I can go to documentation about code monitors by clicking on the link that says “learn more.”
- On the create code monitor page, I can add a name for the code monitor in the input field.
- I cannot change the owner of the code monitor; this is by design.
- On the create and edit code monitor page, I can add a search result trigger to the code monitor using the card.
  - This entails adding a search query in the search query input. The query input will give autocomplete suggestions and syntax highlighting. I can use the “test query” button to open a search results page with this query to verify it works. The search query has certain limitations that must be fulfilled and I can easily know if my query is valid under these limitations.
  - I can confirm this query with the “continue” button
  - I can cancel my changes with the “cancel” button
- If a trigger has been added, I can add actions to the code monitor using the cards.
  - The actions are “send email notifications,” “send Slack message to channel,” and “call a webhook”
  - Each action type opens a new card
- Send email notifications
  - The “recipients” field cannot be changed; this is by design
  - I can send a test email to the email address in the recipients field by using the button labeled “Send test email”
  - I can change the state of the action by using the toggle; the label will reflect the state of the toggle
  - I can use the button labeled “continue” to progress with creating the code monitor
  - I can use the button labeled “cancel” to not add the action
  - I can use the button labeled “delete” to remove the action
- Send Slack message to channel
  - I can click on the link labeled “Slack” to go to the Slack apps website
  - I can go to instructions about setting up Slack messages by clicking on the link labeled “Read more about how to set up Slack webhooks in the docs”
  - I can add a webhook URL in the input labeled “Webhook URL”
  - If a webhook URL has been added, I can send a test message to the Slack channel identified by the webhook URL by using the button labeled “Send test message”
  - I can change the state of the action by using the toggle; the label will reflect the state of the toggle
  - I can use the button labeled “continue” to progress with creating the code monitor
  - I can use the button labeled “cancel” to not add the action
  - I can use the button labeled “delete” to remove the action
- Call a webhook
  - I can add a webhook URL in the input labeled “Webhook URL”
  - If a webhook URL has been added, I can send a test message target identified by the webhook URL by using the button labeled “Call webhook with test payload”
  - I can change the state of the action by using the toggle; the label will reflect the state of the toggle
  - I can use the button labeled “continue” to progress with creating the code monitor
  - I can use the button labeled “cancel” to not add the action
  - I can use the button labeled “delete” to remove the action
- The code monitor can be toggled on or off before saving
  - The default state is “active”
- Once a name, trigger, and action have been added, the code monitor can be saved
  - “Cancel” will not create the code monitor or undo the edits and return to the landing page
- Button labeled “Delete” will delete the code monitor

### How to audit

Follow the instructions here: [Auditing a user journey](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-audit#auditing-a-user-journey)

**Note:** We are planning on arranging thorough third-party audit in the future, so our focus here is just to find the _majority_ of accessibility issues. Don't worry if you feel you haven't found 100% of potential issues, it is better to focus on the core essentials to complete the journey rather than spending lots of time going through every possible problem.

### How to raise discovered problems

Follow the instructions here: [Raising an accessibility bug](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-audit#raising-a-bug)

### Any other questions?

See the [audit FAQ](https://docs.google.com/document/d/1nCi7TvYRpj9eG0_QtgqK1c4L28EiJKVuxyuvg1lAGgo/edit#) - add your question if it isn't there already!

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by limitedmage** [comment]
Audit completed with the following follow-up issues:

- https://github.com/sourcegraph/sourcegraph/issues/35729
- https://github.com/sourcegraph/sourcegraph/issues/35730
- https://github.com/sourcegraph/sourcegraph/issues/35732
- https://github.com/sourcegraph/sourcegraph/issues/35733
- https://github.com/sourcegraph/sourcegraph/issues/35734
- https://github.com/sourcegraph/sourcegraph/issues/35736
- https://github.com/sourcegraph/sourcegraph/issues/35750
